spec/helpers/application_helper/buttons/new_host_aggregate_spec.rb

```diff
@@ -1,19 +1,38 @@
 describe ApplicationHelper::Button::NewHostAggregate do
+  include Spec::Support::RBACHelper
+
   let(:view_context) { setup_view_context_with_sandbox({}) }
-  let(:button) { described_class.new(view_context, {}, {}, {}) }
+  let(:button)       { described_class.new(view_context, {}, {}, {}) }
+
+  before do
+    setup_user_with_tenant
+  end
+
+  let!(:provider) { FactoryGirl.create(:ems_openstack, :tenant => @tenant) }
 
   describe '#disabled?' do
     subject { button[:title] }
 
     context 'no provider available' do
-      before { button.calculate_properties }
+      before do
+        provider.destroy
+        button.calculate_properties
+      end
+
+      it_behaves_like 'a disabled button', "No cloud providers support creating host aggregates."
 
-      it_behaves_like 'a disabled button'
+      context 'when provider is not available due to RBAC rules' do
```

Comment:
`when` is unnecessary, can be assumed from the context

Reply:
but you can also go with `due to RBAC rules` as the context will be concatenated with the previous one